presto-main/src/main/java/com/facebook/presto/sql/planner/plan/ExchangeNode.java

```diff
@@ -62,14 +64,17 @@
     // for each source, the list of inputs corresponding to each output
     private final List<List<Symbol>> inputs;
 
+    private final Optional<OrderingScheme> orderingScheme;
+
```

Comment:
The more I think about it, the less convinced I become that this `orderingScheme` should belong to `ExchangeNode`. IMO it makes `ExchangeNode` (and all the related operators) to be somehow overwhelmed with responsibilities.

Is not that `ExchangeNode` should preserve the order of rows (FIFO)? If so why not to sort before the exchange. That way you would not need change this class at all. What do you think?